src/main/java/com/ldtteam/structurize/client/gui/WindowExtendedBuildTool.java

```diff
@@ -114,21 +151,59 @@ public enum ButtonType
         Back
     }
 
+    public WindowExtendedBuildTool(
+      final BlockPos pos,
+      final int groundstyle)
+    {
+        this(pos, groundstyle, null, BLOCK_BLUEPRINT_REQUIREMENT);
+    }
+
     /**
      * Creates a window build tool.
      * This requires X, Y and Z coordinates.
      * If a structure is active, recalculates the X Y Z with offset.
      * Otherwise the given parameters are used.
      *
-     * @param pos coordinate.
-     * @param groundstyle one of the GROUNDSTYLE_ values.
+     * @param pos                         coordinate.
+     * @param groundstyle                 one of the GROUNDSTYLE_ values.
+     * @param selectionCallback           callback which gets triggered on confirm clicked if a blueprint is selected
+     * @param availableBlueprintPredicate
      */
-    public WindowExtendedBuildTool(@Nullable final BlockPos pos, final int groundstyle)
+    public WindowExtendedBuildTool(
+      final BlockPos pos,
+      final int groundstyle,
+      @Nullable final BiConsumer<WindowExtendedBuildTool, Blueprint> selectionCallback,
+      @Nullable final Predicate<Blueprint> availableBlueprintPredicate)
     {
-        super(MOD_ID + BUILD_TOOL_RESOURCE_SUFFIX, pos, groundstyle,"blueprint");
+        super(MOD_ID + BUILD_TOOL_RESOURCE_SUFFIX, pos, groundstyle, "blueprint");
+        this.selectionCallback = selectionCallback;
+        this.availableBlueprintPredicate = availableBlueprintPredicate;
```

Comment:
I'm not sure if this is the case you're thinking of or a different one, but for Minecolonies I was thinking it would be useful just to extend the `IRequirementsBlueprintAnchorBlock` predicate rather than completely replace it, having a distinction between "this blueprint should be entirely hidden", "this blueprint should be visible but not allow placement", and "this blueprint should show a warning but still allow placement" (and there being no problems, of course).

In particular, for MineColonies, when you don't have the hut block it would still be nice to see the blueprint but not be able to place it; while not having the research but having the hut block would show a warning but still allow placement.

That would restore some pre-1.19 behaviour that's currently unavailable.  I'd imagine this would either mean a function that returns an enum, or multiple functions/predicates to check the different cases.

Reply:
different usecase ye, I want to use it as schematic selector UI in scan features